### PR TITLE
Added task list listener and other edits

### PR DESCRIPTION
**Pull request by katsutori**
Added errors array to users route
Remove conditional in index.js for contacts
Changed tasks.js /post router and removed csurf
Added taskRouter to app.js
Redirect user to /app after login

### CODE REVIEW

**Comment by katsutori** [comment]
Refactored app.pug
Added login conditional to access /app
Added temporary Logout button to app.pug
Updated app.css to include logout button and styling for added tasks

**Comment by katsutori** [comment]
added favicon and completed splash page

**Comment by katsutori** [comment]
Added contacts via backend to the drop down in app.pug. Modified routes/index.js

File: public/javascripts/app.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    let isCompleted;
+    let csrf;
+
+    //convert checkbox to boolean value
+    if (checkStatus === 'on') {
+       isCompleted = true;
+    } else {
+        isCompleted = false;
+    }
+
```
Comment (choi-jihoon) [inline public/javascripts/app.js:59]: This parse-ing out the csrf token shouldn't be necessary if it's included in the form field as a hidden value.

File: public/javascripts/app.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    const body = { description, dueDate, isCompleted, givenTo }
+
+    try {
+        const res = await fetch("http://localhost:8080/tasks", {
+            method: "POST",
+            body: JSON.stringify(body),
+            headers: {
+                "Content-Type": "application/json",
+                'X-CSRF-TOKEN': csrf
```
Comment (choi-jihoon) [inline public/javascripts/app.js]: I don't believe this csrf header is necessary?

File: public/javascripts/app.js (hunk 1 above)
Comment (katsutori) [reply public/javascripts/app.js:59]: Yep, you are correct. I talked to Alec about this, but we couldn't get the csrf to pass one way or the other. We thought that parsing it and passing through the body would work, but that didn't work either.

File: public/javascripts/utils.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+export const handleErrors = async (err) => {
+    if (err.status >= 400 && err.status < 600) {
+      const errorJSON = await err.json();
+      const errorsContainer = document.querySelector(".errors-container");
+      let errorsHtml = [
+        `
+          <div class="alert alert-danger">
```
Comment (choi-jihoon) [inline public/javascripts/utils.js:7]: Is this class a bootstrap or a Thienstrap?

File: public/javascripts/utils.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+              Something went wrong. Please try again.
+          </div>
+        `,
+      ];
+      const { errors } = errorJSON;
+      if (errors && Array.isArray(errors)) {
+        errorsHtml = errors.map(
+          (message) => `
+            <div class="alert alert-danger">
```
Comment (choi-jihoon) [inline public/javascripts/utils.js:16]: (see above, just making sure! My CSS knowledge is lacking

File: routes/index.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 const express = require('express');
 const { csrfProtection, asyncHandler, handleValidationErrors } = require("../utils");
 const db = require('../db/models');
-const { Contact } = db;
+const { Contact, Task } = db;
 
 var router = express.Router();
```
Comment (choi-jihoon) [inline routes/index.js:7]: Let's change this to const later!

File: routes/tasks.js [hunk 6]
```diff
@@ -30,13 +30,38 @@ router.get('/users/:id(\\d+)/tasks', csrfProtection, asyncHandler(async(req, res
...
+
+router.post('/', validateTask, handleValidationErrors, asyncHandler(async(req, res) => {
+  const { description, dueDate, isCompleted, givenTo } = req.body;
   const userId = res.locals.userId
   const task = await Task.create({
     userId,
     description,
     dueDate,
+    isCompleted,
     givenTo
```
Comment (choi-jihoon) [inline routes/tasks.js:65]: We originally didn't add an isCompleted as a form requirement because the default value of a task is false

**Review by choi-jihoon — APPROVED** [review]

**Review by strewm — APPROVED** [review]

**Review by Codetrauma — APPROVED** [review]

**Pull request merged**